Some minor copy edit on sandbox documentation

ColinEberhardt opened the pull request.

tarruda commented on docs/api/sandbox-option.md:
```diff
@@ -1,9 +1,9 @@
 # `sandbox` Option
 
-> Create a browser window with renderer that can run inside chromium OS sandbox.
+> Create a browser window with a renderer that runs inside the Chromium OS sandbox.
```
Passing `sandbox` alone does not enable chromium os sandbox, it simply creates a renderer that is compatible with it. In other words, a sandbox-compatible renderer will only communicate with system using IPC through main process.(read about `--enable-sandbox` option below)

ColinEberhardt replied:
Thanks for the clarification, that was something that is a little unclear in your documentation. 

I think this is something that needs to be clearly stated up-front. I know people who are using `sandbox: true` under the impression that the Chromium Sandbox is being used.

I'll do another quick edit.

ColinEberhardt replied:
I've updated the summary to be more comprehensive: https://github.com/electron/electron/pull/9054/commits/92d33995372510b2a6c93c4aa7bda5860ecfa3d2

As an aside, I do wonder if there is much value in using `sandbox: true` without `--enable-sandbox`? And the very existence of this option makes it quite easy to accidentally think that you have the Chromium sandbox enabled when in fact it isn't.

Perhaps the property should be renamed `sandbox-compatible: true`?

tarruda replied:
> As an aside, I do wonder if there is much value in using sandbox: true without --enable-sandbox?

As far as I can see, there are 3 use cases for `sandbox` option without `--enable-sandbox`:

- It is the only way to get native `window.open` behavior(This may change with #8963 though).
- It stops electron from creating a node.js runtime in the renderer, which is good from a security POV since it potentially reduces attack surface.
- For linux(and possibly windows too) you can't obtain crash dumps if `--enable-sandbox` is used because breakpad requires filesystem access to write the dump file. For now the user has to remove `--enable-sandbox` in order to debug crashes, though this may change in the future if electron starts using crashpad for all platforms. (BTW, feel free to include this information in the doc 😄 )

> Perhaps the property should be renamed sandbox-compatible: true?

Considering some people are already using the option, it is better keep the name for now. Maybe in electron 2.0.

ColinEberhardt commented:
@tarruda rebased and squashed. I also added a few additional points based on your [earlier comment](https://github.com/electron/electron/pull/9054#discussion_r108727588)

tarruda commented:
Anyone else want to review these changes? cc @zeke @kevinsawicki

kevinsawicki commented:
Thanks for this @ColinEberhardt 👍 🚀

ColinEberhardt commented:
Thanks everyone :-)
